scripts/sendpayment.py

```diff
@@ -115,6 +115,17 @@ def main():
             options.txfee))
     assert (options.txfee >= 0)
 
+    # From the estimated tx fees, check if the expected amount is a
+    # significant value compared the the cj amount
+    exp_tx_fees_ratio = ((1 + options.makercount) * options.txfee) / amount
```

Comment:
This is certainly a reasonable idea, but are we now checking both percentage and absolute sanity checks for fees, with different code? I think so. I think this area needs significant rationalisation (look at the use of txfee_default in Taker.py ; I actually see one minor error there already,  in the P2EP code, but leave that aside, there is still a need for rationalisation how the pre-tx-construction estimation works).

Reply:
Yes, you're correct it's different code checking for absolute and relative fees.

Though I'd argue that the absolute check is just the configurable value `absurd_fee_per_kb` from the joinmarket.cfg file. That's arguably just there to prevent excesses in the fee market or joinmarket misconfigurations. But it won't help at all against the issue we've seen sometimes, that takers use joinmarket for very small amounts and are shocked to find that a huge percentage of their funds went to fees (which were well below the `absurd_fee_per_kb` amount).

While we could argue that we can just get rid of the `absurd_fee_per_kb` setting if this PR is merged, I'd beg to differ: in the extreme case, e.g. when I wanted to mix 100 BTC, a misconfiguration might lead to set the tx fee to something like 500000 sat/kbyte which will not be caught by this PR (still too small, percentage-wise) but would be caught by the absurd fee setting.

Or did I maybe misunderstand your point? I'm happy to re-check any tx_fee safeguards and re-write the PR.